### PR TITLE
Gitops basic tutorial update

### PR DESCRIPTION
**Pull request by carillan81**
### What does this PR do?

Update gitops basic guide including:
 - kubectl gs gitops commands
 - gitops-template repo structure
 - sops encryption

### What does it look like?

Apart from the first part of the doc that explains concepts, all the practical part has been replaced.

### Any background context you can provide?

Part of this issue:
https://github.com/giantswarm/roadmap/issues/1624

### What is needed from the reviewers?
Ensure that the doc has no errors and that the process is explained so users can follow

### Have you maintained the front matter?
Yes, updated review date.
(Please bump the last_review_date in case this qualifies as a review for an entire page. Provide user_questions which should be answered in the page. Provide a meaningful description.)

### CODE REVIEW

File: src/content/advanced/gitops/index.md
Comment (pipo02mix) [inline src/content/advanced/gitops/index.md:128]:
```suggestion
A GitOps repository can have any arbitrary structure. This provides flexibility to adapt the repo to your own infra structure. In Giant Swarm, we have a clear hierarchical structure and we recommend using this structure. This provides some advantages:
```

File: src/content/advanced/gitops/index.md
Comment (pipo02mix) [inline src/content/advanced/gitops/index.md:130]:
```suggestion
- It's easy to locate resources and know what and where is being deployed
```

File: src/content/advanced/gitops/index.md
Comment (pipo02mix) [inline src/content/advanced/gitops/index.md:157]:
```suggestion
We provide a template repository where this structure is elaborated and explained in detail in [giantswarm/gitops-template](https://github.com/giantswarm/gitops-template). It is not necessary to clone the repository to follow this guide, but it can provide a good reference for every possible use case. We recommend copying, at least, the bases folder from that repository to your own repository because cluster creation uses them as templates as we will see later in this document.

```

File: src/content/advanced/gitops/index.md
Comment (pipo02mix) [inline src/content/advanced/gitops/index.md:160]:
```suggestion
First things first - create a bare-bones repository where we will store the resources that will be deployed. In this repository, we need to create the structure described in the last section and copy the bases for workload clusters and other apps in the Giant Swarm catalog. Follow these steps:
```

File: src/content/advanced/gitops/index.md
Comment (pipo02mix) [inline src/content/advanced/gitops/index.md:170]:
```suggestion
__Note__: To learn more about Giant Swarm's kubectl plugin, visit [kubectl-gs documentation]({{< relref "/ui-api/kubectl-gs/" >}}).
```
We use this syntax

File: src/content/advanced/gitops/index.md
Comment (pipo02mix) [inline src/content/advanced/gitops/index.md:180]:
```suggestion
This command creates the folder for the management cluster, the `Kustomization` file for the MC, `organizations` folder and `secrets` folder, already populated with an encoding key.
```

File: src/content/advanced/gitops/index.md
Comment (pipo02mix) [inline src/content/advanced/gitops/index.md:182]:
```suggestion
Then, we create a Flux `GitRepository` resource that points to our repository. We need a `GitRepository` resource for every management cluster we want to manage. In the `GitRepository` resource we define the URL where our repository is stored.
```
Can we add a GitRepository example?

File: src/content/advanced/gitops/index.md
Comment (pipo02mix) [inline src/content/advanced/gitops/index.md:185]:
```suggestion
First, we need to create a secret (we will refer to it as `GIT_CREDENTIALS_SECRET`, please, replace it with the chosen name) with the credentials to access the repository. This will allow Flux to download the repo files in order to apply all resources. If your repo is public you don't need the secret and can safely delete the parameter `secret-ref` of the next command.

This process is different depending on the git platform used (ssh-keys, token, user/password). Check the documentation of every provider to create the secret.
```

File: src/content/advanced/gitops/index.md
Comment (pipo02mix) [inline src/content/advanced/gitops/index.md:198]:
```suggestion
This command creates the Custom Resource for us and exports it to a YAML file in the management-cluster folder.
```

File: src/content/advanced/gitops/index.md
Comment (pipo02mix) [inline src/content/advanced/gitops/index.md:226]:
```suggestion
The key was already created for us in the `kubectl gs gitops add management-cluster` command so we only need to apply the secret in the cluster and safely store the private GPG key so we can use it to encrypt files before pushing them to the repo.
```
Instead of put the command we could add an anchor to the step

File: src/content/advanced/gitops/index.md
Comment (pipo02mix) [inline src/content/advanced/gitops/index.md:242]:
```suggestion
Now, in order to safely store the key in the repo we can encode the YAML file using `sops`:
```

File: src/content/advanced/gitops/index.md
Comment (pipo02mix) [inline src/content/advanced/gitops/index.md:258]:
```suggestion
In order to create the files in the repository, run the command (in the root of the repo):
```

File: src/content/advanced/gitops/index.md
Comment (pipo02mix) [inline src/content/advanced/gitops/index.md:264]:
```suggestion
This is enough to create the Organization on our own. It creates automatically the organization namespace where your cluster resources will be applied.
```

File: src/content/advanced/gitops/index.md
Comment (pipo02mix) [inline src/content/advanced/gitops/index.md:275]:
```suggestion
In the next step, we have the organization created so now we can define some workload clusters. In Giant Swarm, we have developed several providers where clusters can be provisioned. Every provider has a different syntax for the infrastructure definition and to reduce the complexity we have defined some [`bases`](https://github.com/giantswarm/gitops-template/tree/main/bases) that helps to configure the clusters.
```

File: src/content/advanced/gitops/index.md
Comment (pipo02mix) [inline src/content/advanced/gitops/index.md:277]:
```suggestion
The configuration structures in layers so every upper layer can modify the configuration and create a custom and very powerful structure. The different possible layers would be:
```

File: src/content/advanced/gitops/index.md
Comment (pipo02mix) [inline src/content/advanced/gitops/index.md:282]:
```suggestion
- Base with fundamental cluster creation manifests
- Base versions with different modifications
- Environments that implement config modifications in the bases
- Clusters that implement specific configurations
```

File: src/content/advanced/gitops/index.md
Comment (pipo02mix) [inline src/content/advanced/gitops/index.md:284]:
```suggestion
There are a bunch of advantages to createing clusters starting from a base:
```

File: src/content/advanced/gitops/index.md
Comment (pipo02mix) [inline src/content/advanced/gitops/index.md:288]:
```suggestion
- We can group clusters in logical groups that match our infra (dev, pre, prod...)
- Modifying the base, we modify all the clusters that implement it (batch update)
- We can change the clusters between different environments easily
```

File: src/content/advanced/gitops/index.md
Comment (pipo02mix) [inline src/content/advanced/gitops/index.md:290]:
```suggestion
In this tutorial, we are implementing the simplest form, which is, to implement a cluster from a base without applying any extra configuration.
```

File: src/content/advanced/gitops/index.md
Comment (pipo02mix) [inline src/content/advanced/gitops/index.md:300]:
```suggestion
You can add user configurations to the cluster created including different config maps and add them to the manifests.
```

File: src/content/advanced/gitops/index.md
Comment (pipo02mix) [inline src/content/advanced/gitops/index.md:304]:
```suggestion
It is just as easy to install managed apps in existing workload clusters. In this part of the guide, we will assume you have completed the previous steps and have both an organization and a cluster running.
```

File: src/content/advanced/gitops/index.md
Comment (pipo02mix) [inline src/content/advanced/gitops/index.md:308]:
```suggestion
We are showing the process of installing `Grafana` from the `GiantSwarm` app catalog in the namespace `monitoring`. In order to add the app files to the repo, run the command (check version numbers for the latest releases):

```

File: src/content/advanced/gitops/index.md
Comment (pipo02mix) [inline src/content/advanced/gitops/index.md:313]:
```suggestion
The above command has created the files and folders needed and the app will be deployed as these files are committed and pushed.
```

**Review by pipo02mix — COMMENTED** [review]
I did not try it, just content review. I will try to follow next week and provide more feedback

File: src/content/advanced/gitops/index.md
Comment (mproffitt) [inline src/content/advanced/gitops/index.md:224]:
```suggestion
In this example we are using `sops` with `pgp` key management and creating a master key for all the kustomizations in this management cluster.
```

File: src/content/advanced/gitops/index.md
Comment (mproffitt) [inline src/content/advanced/gitops/index.md:226]:
```suggestion
This master key is created automatically by the `kubectl gs gitops add management-cluster` command so we only need to apply the secret in the cluster and safely store the private gpg key so we can use it to encrypt files before pushing them to the repo.
```

File: src/content/advanced/gitops/index.md
Comment (mproffitt) [inline src/content/advanced/gitops/index.md:274]:
```suggestion
Once we have an organization created, we can create workload clusters inside it to run our applications.
```

File: src/content/advanced/gitops/index.md
Comment (mproffitt) [inline src/content/advanced/gitops/index.md:277]:
```suggestion
The configuration is structured in such a way that each layer can modify the configuration and create a custom and very powerful structure. The different possible layers would be:
```

File: src/content/advanced/gitops/index.md
Comment (mproffitt) [inline src/content/advanced/gitops/index.md:290]: I re-wrote the term `simplest form` under SOPS, I think this is another term I don't like (pedant of me, I know)

```suggestion
In this tutorial we are implementing a cluster from a base without applying any extra configuration.
```

File: src/content/advanced/gitops/index.md
Comment (mproffitt) [inline src/content/advanced/gitops/index.md:307]:
```suggestion
In order to generate the manifests for installing our `Grafana` managed app from the the `giantswarm` app-catalog into the `monitoring` namespace, we run the command
```

File: src/content/advanced/gitops/index.md
Comment (mproffitt) [inline src/content/advanced/gitops/index.md:313]:
```suggestion
The latest version number can be retrieved from the catalog using `helm search repo giantswarm/grafana`

Commit the newly generated files back to your git repository and push the changes to the remote. After a few minutes you should see the application resources appear on your workload cluster.
```

**Review by mproffitt — CHANGES_REQUESTED** [review]

**Comment by marians** [comment]
I'd like to review this one as soon as suggestions are applied. :)

File: src/content/advanced/gitops/index.md
Comment (mproffitt) [inline src/content/advanced/gitops/index.md:183]:
```suggestion
Now, we need to create a secret (we will refer to it as GIT_CREDENTIALS_SECRET, please, replace with the chosen name) with the credentials to access the repository. This will allow Flux to download the repo files in order to apply all resources. If your repo is public you don't need the secret and can safely delete the parameter `secret-ref` of the next command.
```

File: src/content/advanced/gitops/index.md
Comment (mproffitt) [inline src/content/advanced/gitops/index.md:255]:
```suggestion
The highest level Giant Swarm custom resource you can create is an [Organization]({{< relref "/ui-api/management-api/crd/organizations.security.giantswarm.io.md" >}}).
```

**Review by mproffitt — CHANGES_REQUESTED** [review]
Minor spelling/grammar suggestions

File: src/content/advanced/gitops/index.md
Comment (pipo02mix) [inline src/content/advanced/gitops/index.md:132]:
```suggestion
- The general layout of the repository closely mirrors the infrastructure defined within Giant Swarm.
- It's easy to locate resources and know what they are and where they are deployed.
- Our engineers are familiar with the layout which saves time during support requests.
```

File: src/content/advanced/gitops/index.md
Comment (pipo02mix) [reply src/content/advanced/gitops/index.md:159]: @carillan81 I think you miss this one

File: src/content/advanced/gitops/index.md
Comment (pipo02mix) [inline src/content/advanced/gitops/index.md:197]:
```suggestion
This command creates the Custom Resource for us and exports it to a YAML file in the `management-clusters` folder.
```

File: src/content/advanced/gitops/index.md
Comment (pipo02mix) [inline src/content/advanced/gitops/index.md:221]:
```suggestion
Next, we configure the keys that will be used in the management cluster Flux to decipher secrets so they can be safely stored in the repo.
```
Please be consistent with uppercase handling

File: src/content/advanced/gitops/index.md
Comment (pipo02mix) [reply src/content/advanced/gitops/index.md:282]: this is missing too

File: src/content/advanced/gitops/index.md
Comment (pipo02mix) [reply src/content/advanced/gitops/index.md:288]: other one

File: src/content/advanced/gitops/index.md
Comment (pipo02mix) [inline src/content/advanced/gitops/index.md:273]: I dont understand this paragraph here. Which managed application?

**Comment by pipo02mix** [comment]
Some comments were not yet replied and I added some other suggestions.

It is hard to see how it will look after all these changes. I have had to use `View File` option. I think the steps are explained in the right order with the right commands but I miss a bit of context on why the user is doing the action. Example [here](https://github.com/giantswarm/docs/pull/1637/files#diff-50aa53b33584f1f957cb3380fecb9da060a61179de124426a466add81a2a039fR221) you talk about the keys first time assuming readers knows what those are about. My suggestion would be to introduce the purpose "In GitOps, secrets are usually stored together with the rest of the configuration. For security implications, even if our git repository is private, the secrets must be encrypted. For that reason we use GPG .... " WDYT?

File: src/content/advanced/gitops/index.md
Comment (mproffitt) [reply src/content/advanced/gitops/index.md:273]: This paragraph looks like overflow - managed apps are discussed in the section below, I think this can/should be either deleted or moved to the "Installing managed apps" section.

File: src/content/advanced/gitops/index.md
Comment (carillan81) [reply src/content/advanced/gitops/index.md:159]: I changed the term `first` for other wording in those cases

**Comment by carillan81** [comment]
> Some comments were not yet replied and I added some other suggestions.
> 
> It is hard to see how it will look after all these changes. I have had to use `View File` option. I think the steps are explained in the right order with the right commands but I miss a bit of context on why the user is doing the action. Example [here](https://github.com/giantswarm/docs/pull/1637/files#diff-50aa53b33584f1f957cb3380fecb9da060a61179de124426a466add81a2a039fR221) you talk about the keys first time assuming readers knows what those are about. My suggestion would be to introduce the purpose "In GitOps, secrets are usually stored together with the rest of the configuration. For security implications, even if our git repository is private, the secrets must be encrypted. For that reason we use GPG .... " WDYT?

As this article is pretty outdated I would rather merge an imperfect version and address this issue in another iteration than block the merge.

File: src/content/advanced/gitops/index.md
Comment (carillan81) [reply src/content/advanced/gitops/index.md]: this has been addressed

**Review by pipo02mix — APPROVED** [review]
LGTM as first version if other reviewers are happy

**Pull request merged**
